<!DOCTYPE html>

<html>

<head>

<meta charset="utf-8" />
<meta name="generator" content="pandoc" />
<meta http-equiv="X-UA-Compatible" content="IE=EDGE" />

<meta name="viewport" content="width=device-width, initial-scale=1" />



<title>Case study: the true finches</title>

<script>// Pandoc 2.9 adds attributes on both header and div. We remove the former (to
// be compatible with the behavior of Pandoc < 2.8).
document.addEventListener('DOMContentLoaded', function(e) {
  var hs = document.querySelectorAll("div.section[class*='level'] > :first-child");
  var i, h, a;
  for (i = 0; i < hs.length; i++) {
    h = hs[i];
    if (!/^h[1-6]$/i.test(h.tagName)) continue;  // it should be a header h1-h6
    a = h.attributes;
    while (a.length > 0) h.removeAttribute(a[0].name);
  }
});
</script>

<style type="text/css">
code{white-space: pre-wrap;}
span.smallcaps{font-variant: small-caps;}
span.underline{text-decoration: underline;}
div.column{display: inline-block; vertical-align: top; width: 50%;}
div.hanging-indent{margin-left: 1.5em; text-indent: -1.5em;}
ul.task-list{list-style: none;}
</style>



<style type="text/css">
code {
white-space: pre;
}
.sourceCode {
overflow: visible;
}
</style>
<style type="text/css" data-origin="pandoc">
pre > code.sourceCode { white-space: pre; position: relative; }
pre > code.sourceCode > span { display: inline-block; line-height: 1.25; }
pre > code.sourceCode > span:empty { height: 1.2em; }
.sourceCode { overflow: visible; }
code.sourceCode > span { color: inherit; text-decoration: inherit; }
div.sourceCode { margin: 1em 0; }
pre.sourceCode { margin: 0; }
@media screen {
div.sourceCode { overflow: auto; }
}
@media print {
pre > code.sourceCode { white-space: pre-wrap; }
pre > code.sourceCode > span { text-indent: -5em; padding-left: 5em; }
}
pre.numberSource code
{ counter-reset: source-line 0; }
pre.numberSource code > span
{ position: relative; left: -4em; counter-increment: source-line; }
pre.numberSource code > span > a:first-child::before
{ content: counter(source-line);
position: relative; left: -1em; text-align: right; vertical-align: baseline;
border: none; display: inline-block;
-webkit-touch-callout: none; -webkit-user-select: none;
-khtml-user-select: none; -moz-user-select: none;
-ms-user-select: none; user-select: none;
padding: 0 4px; width: 4em;
color: #aaaaaa;
}
pre.numberSource { margin-left: 3em; border-left: 1px solid #aaaaaa; padding-left: 4px; }
div.sourceCode
{ }
@media screen {
pre > code.sourceCode > span > a:first-child::before { text-decoration: underline; }
}
code span.al { color: #ff0000; font-weight: bold; } 
code span.an { color: #60a0b0; font-weight: bold; font-style: italic; } 
code span.at { color: #7d9029; } 
code span.bn { color: #40a070; } 
code span.bu { color: #008000; } 
code span.cf { color: #007020; font-weight: bold; } 
code span.ch { color: #4070a0; } 
code span.cn { color: #880000; } 
code span.co { color: #60a0b0; font-style: italic; } 
code span.cv { color: #60a0b0; font-weight: bold; font-style: italic; } 
code span.do { color: #ba2121; font-style: italic; } 
code span.dt { color: #902000; } 
code span.dv { color: #40a070; } 
code span.er { color: #ff0000; font-weight: bold; } 
code span.ex { } 
code span.fl { color: #40a070; } 
code span.fu { color: #06287e; } 
code span.im { color: #008000; font-weight: bold; } 
code span.in { color: #60a0b0; font-weight: bold; font-style: italic; } 
code span.kw { color: #007020; font-weight: bold; } 
code span.op { color: #666666; } 
code span.ot { color: #007020; } 
code span.pp { color: #bc7a00; } 
code span.sc { color: #4070a0; } 
code span.ss { color: #bb6688; } 
code span.st { color: #4070a0; } 
code span.va { color: #19177c; } 
code span.vs { color: #4070a0; } 
code span.wa { color: #60a0b0; font-weight: bold; font-style: italic; } 
</style>
<script>
// apply pandoc div.sourceCode style to pre.sourceCode instead
(function() {
  var sheets = document.styleSheets;
  for (var i = 0; i < sheets.length; i++) {
    if (sheets[i].ownerNode.dataset["origin"] !== "pandoc") continue;
    try { var rules = sheets[i].cssRules; } catch (e) { continue; }
    var j = 0;
    while (j < rules.length) {
      var rule = rules[j];
      // check if there is a div.sourceCode rule
      if (rule.type !== rule.STYLE_RULE || rule.selectorText !== "div.sourceCode") {
        j++;
        continue;
      }
      var style = rule.style.cssText;
      // check if color or background-color is set
      if (rule.style.color === '' && rule.style.backgroundColor === '') {
        j++;
        continue;
      }
      // replace div.sourceCode by a pre.sourceCode rule
      sheets[i].deleteRule(j);
      sheets[i].insertRule('pre.sourceCode{' + style + '}', j);
    }
  }
})();
</script>




<style type="text/css">body {
background-color: #fff;
margin: 1em auto;
max-width: 700px;
overflow: visible;
padding-left: 2em;
padding-right: 2em;
font-family: "Open Sans", "Helvetica Neue", Helvetica, Arial, sans-serif;
font-size: 14px;
line-height: 1.35;
}
#TOC {
clear: both;
margin: 0 0 10px 10px;
padding: 4px;
width: 400px;
border: 1px solid #CCCCCC;
border-radius: 5px;
background-color: #f6f6f6;
font-size: 13px;
line-height: 1.3;
}
#TOC .toctitle {
font-weight: bold;
font-size: 15px;
margin-left: 5px;
}
#TOC ul {
padding-left: 40px;
margin-left: -1.5em;
margin-top: 5px;
margin-bottom: 5px;
}
#TOC ul ul {
margin-left: -2em;
}
#TOC li {
line-height: 16px;
}
table {
margin: 1em auto;
border-width: 1px;
border-color: #DDDDDD;
border-style: outset;
border-collapse: collapse;
}
table th {
border-width: 2px;
padding: 5px;
border-style: inset;
}
table td {
border-width: 1px;
border-style: inset;
line-height: 18px;
padding: 5px 5px;
}
table, table th, table td {
border-left-style: none;
border-right-style: none;
}
table thead, table tr.even {
background-color: #f7f7f7;
}
p {
margin: 0.5em 0;
}
blockquote {
background-color: #f6f6f6;
padding: 0.25em 0.75em;
}
hr {
border-style: solid;
border: none;
border-top: 1px solid #777;
margin: 28px 0;
}
dl {
margin-left: 0;
}
dl dd {
margin-bottom: 13px;
margin-left: 13px;
}
dl dt {
font-weight: bold;
}
ul {
margin-top: 0;
}
ul li {
list-style: circle outside;
}
ul ul {
margin-bottom: 0;
}
pre, code {
background-color: #f7f7f7;
border-radius: 3px;
color: #333;
white-space: pre-wrap; 
}
pre {
border-radius: 3px;
margin: 5px 0px 10px 0px;
padding: 10px;
}
pre:not([class]) {
background-color: #f7f7f7;
}
code {
font-family: Consolas, Monaco, 'Courier New', monospace;
font-size: 85%;
}
p > code, li > code {
padding: 2px 0px;
}
div.figure {
text-align: center;
}
img {
background-color: #FFFFFF;
padding: 2px;
border: 1px solid #DDDDDD;
border-radius: 3px;
border: 1px solid #CCCCCC;
margin: 0 5px;
}
h1 {
margin-top: 0;
font-size: 35px;
line-height: 40px;
}
h2 {
border-bottom: 4px solid #f7f7f7;
padding-top: 10px;
padding-bottom: 2px;
font-size: 145%;
}
h3 {
border-bottom: 2px solid #f7f7f7;
padding-top: 10px;
font-size: 120%;
}
h4 {
border-bottom: 1px solid #f7f7f7;
margin-left: 8px;
font-size: 105%;
}
h5, h6 {
border-bottom: 1px solid #ccc;
font-size: 105%;
}
a {
color: #0033dd;
text-decoration: none;
}
a:hover {
color: #6666ff; }
a:visited {
color: #800080; }
a:visited:hover {
color: #BB00BB; }
a[href^="http:"] {
text-decoration: underline; }
a[href^="https:"] {
text-decoration: underline; }

code > span.kw { color: #555; font-weight: bold; } 
code > span.dt { color: #902000; } 
code > span.dv { color: #40a070; } 
code > span.bn { color: #d14; } 
code > span.fl { color: #d14; } 
code > span.ch { color: #d14; } 
code > span.st { color: #d14; } 
code > span.co { color: #888888; font-style: italic; } 
code > span.ot { color: #007020; } 
code > span.al { color: #ff0000; font-weight: bold; } 
code > span.fu { color: #900; font-weight: bold; } 
code > span.er { color: #a61717; background-color: #e3d2d2; } 
</style>




</head>

<body>




<h1 class="title toc-ignore">Case study: the true finches</h1>



<p>The family Fringillidae is a taxon that includes the true finches.
Let’s see what is known about their time of divergence.</p>
<div id="getting-species-names-within-the-family-fringillidae" class="section level2">
<h2>1. Getting species names within the family Fringillidae</h2>
<p>DateLife searches its chronogram database using species names
standardized to a <a href="https://tree.opentreeoflife.org/about/taxonomy-version/ott3.3">unified
taxonomy</a> constructed by the Open Tree of Life project.</p>
<p>When working with taxon names above the species level in DateLife,
the first step is to obtain all species names belonging to the taxon of
interest.</p>
<p>We will get all species names belonging to the family Fringillidae,
using the flag <code>get_spp_from_taxon = TRUE</code> from the function
<code>datelife::make_datelife_query()</code>:</p>
<div class="sourceCode" id="cb1"><pre class="sourceCode r"><code class="sourceCode r"><span id="cb1-1"><a href="#cb1-1" aria-hidden="true" tabindex="-1"></a>dquery <span class="ot">&lt;-</span> datelife<span class="sc">::</span><span class="fu">make_datelife_query</span>(<span class="at">input =</span> <span class="st">&quot;fringillidae&quot;</span>,               </span>
<span id="cb1-2"><a href="#cb1-2" aria-hidden="true" tabindex="-1"></a>                                   <span class="at">get_spp_from_taxon =</span> <span class="cn">TRUE</span>)</span>
<span id="cb1-3"><a href="#cb1-3" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb1-4"><a href="#cb1-4" aria-hidden="true" tabindex="-1"></a><span class="co">#&gt; ... Making a DateLife query.</span></span>
<span id="cb1-5"><a href="#cb1-5" aria-hidden="true" tabindex="-1"></a><span class="co">#&gt; ... Phylo-processing &#39;input&#39;.</span></span>
<span id="cb1-6"><a href="#cb1-6" aria-hidden="true" tabindex="-1"></a><span class="co">#&gt; &#39;input&#39; is not a phylogeny.</span></span>
<span id="cb1-7"><a href="#cb1-7" aria-hidden="true" tabindex="-1"></a><span class="co">#&gt;   |=========================================================| 100%</span></span>
<span id="cb1-8"><a href="#cb1-8" aria-hidden="true" tabindex="-1"></a><span class="co">#&gt;   |=========================================================| 100%</span></span>
<span id="cb1-9"><a href="#cb1-9" aria-hidden="true" tabindex="-1"></a><span class="co">#&gt;   |=========================================================| 100%</span></span>
<span id="cb1-10"><a href="#cb1-10" aria-hidden="true" tabindex="-1"></a><span class="co">#&gt;   |=========================================================| 100%</span></span>
<span id="cb1-11"><a href="#cb1-11" aria-hidden="true" tabindex="-1"></a><span class="co">#&gt;   |=========================================================| 100%</span></span>
<span id="cb1-12"><a href="#cb1-12" aria-hidden="true" tabindex="-1"></a><span class="co">#&gt; Working with the following taxa:</span></span>
<span id="cb1-13"><a href="#cb1-13" aria-hidden="true" tabindex="-1"></a><span class="co">#&gt;  Crithagra sulphurata | Crithagra citrinelloides | Crithagra mozambica | Crithagra capistrata | Crithagra burtoni | Crithagra rufobrunnea | Crithagra flaviventris | Callacanthis burtoni | Chaunoproctus ferreorostris | Rhynchostruthus socotranus | Chlorophonia cyanea | Chlorophonia occipitalis | Chlorophonia flavirostris | Akialoa obscura | Euphonia gouldi | Euphonia plumbea | Euphonia elegantissima | Euphonia luteicapilla | Euphonia chrysopasta | Euphonia fulvicrissa | Euphonia hirundinacea | Euphonia chlorotica | Euphonia cyanocephala | Euphonia rufiventris | Euphonia minuta | Euphonia cayennensis | Euphonia xanthogaster | Euphonia violacea | Euphonia pectoralis | Euphonia musica | Euphonia anneae | Euphonia affinis | Euphonia laniirostris | Euphonia finschi | Psittirostra psittacea | Urocynchramus pylzowi | Telespiza cantans | Telespiza ultima | Palmeria dolei | Oreomystis bairdi | Loxops coccineus | Loxops mana | Loxops caeruleirostris | Himatione sanguinea | Rhodopechys sanguineus | Paroreomyza montana | Pseudonestor xanthophrys | Vestiaria coccinea | Bucanetes githagineus | Hemignathus parvus | Hemignathus wilsoni | Hemignathus virens | Hemignathus munroi | Hemignathus flavus | Hemignathus kauaiensis | Hemignathus stejnegeri | Loxioides bailleui | Eremopsaltria mongolica | Rhodospiza obsoleta | Peucedramus taeniatus | Cyanerpes caeruleus | Cyanerpes cyaneus | Cyanerpes nitidus | Cyanerpes lucidus | Fringillaria goslingi | Fringillaria striolata | Fringillaria impetuani | Fringillaria poliopleura | Fringillaria tahapisi | Hemispingus atropileus | Hemispingus verticalis | Hemispingus auricularis | Hemispingus calophrys | Hemispingus rufosuperciliaris | Hemispingus superciliaris | Hemispingus trifasciatus | Hemispingus piurae | Hemispingus xanthophthalmus | Hemispingus parodii | Hemispingus goeringi | Hemispingus melanotis | Hemispingus frontalis | Hemispingus reyi | Rowettia goughensis | Embernagra platensis | Schoeniclus rusticus | Schoeniclus aureolus | Schoeniclus elegans | Schoeniclus siemsseni | Schoeniclus spodocephala | Schoeniclus yessoensis | Tersina viridis | Orthogonys chloricterus | Buarremon apertus | Buarremon assimilis | Buarremon atricapillus | Buarremon basilicus | Buarremon virenticeps | Buarremon torquatus | Peucaea aestivalis | Peucaea cassinii | Peucaea botterii | Phaeothlypis fulvicauda | Donacospiza albifrons | Platyspiza crassirostris | Melopyrrha nigra | Carduelis citrinella | Carduelis corsicana | Carduelis carduelis | Uragus sibiricus | Melamprosops phaeosoma | Hesperiphona vespertina | Pyrrhoplectes epauletta | Haemorhous mexicanus | Haemorhous purpureus | Haemorhous cassinii | Kozlowia roborowskii | Serinus syriacus | Serinus albogularis | Serinus atrogularis | Serinus citrinipectus | Serinus citrinella | Serinus pusillus | Serinus canaria | Serinus alario | Serinus striolatus | Serinus melanochrous | Serinus whytii | Serinus mennelli | Serinus reichardi | Serinus scotops | Serinus flavivertex | Serinus totta | Serinus estherae | Serinus hypostictus | Serinus leucopygius | Serinus thibetanus | Serinus gularis | Serinus dorsostriatus | Serinus canicollis | Serinus serinus | Linaria flavirostris | Linaria cannabina | Chloris monguilloti | Chloris sinica | Chloris spinoides | Chloris ambigua | Chloris chloris | Spinus yarrellii | Spinus spinus | Spinus olivaceus | Spinus uropygialis | Spinus siemiradzkii | Spinus atriceps | Spinus dominicensis | Spinus xanthogastrus | Spinus magellanicus | Spinus pinus | Spinus notatus | Spinus tristis | Spinus spinescens | Spinus crassirostris | Spinus barbatus | Spinus atratus | Spinus lawrencei | Spinus psaltria | Spinus cucullatus | Neospiza concolor | Loxia curvirostra | Loxia pytyopsittacus | Loxia leucoptera | Loxia scotica | Eophona migratoria | Eophona personata | Haematospiza sipahi | Mycerobas icterioides | Mycerobas melanozanthos | Mycerobas affinis | Mycerobas carnipes | Pinicola subhimachala | Pinicola enucleator | Linurgus olivaceus | Leucosticte sillemi | Leucosticte arctoa | Leucosticte australis | Leucosticte nemoricola | Leucosticte tephrocotis | Leucosticte atrata | Leucosticte brandti | Carpodacus vinaceus | Carpodacus rodopeplus | Carpodacus davidianus | Carpodacus rhodochlamys | Carpodacus rubescens | Carpodacus synoicus | Carpodacus severtzovi | Carpodacus rodochroa | Carpodacus erythrinus | Carpodacus roseus | Carpodacus nipalensis | Carpodacus trifasciatus | Carpodacus rubicilloides | Carpodacus pulcherrimus | Carpodacus thura | Carpodacus eos | Carpodacus rubicilla | Carpodacus grandis | Carpodacus edwardsii | Carpodacus puniceus | Carpodacus formosanus | Coccothraustes coccothraustes | Acanthis hornemanni | Acanthis cabaret | Acanthis flammea | Pyrrhula nipalensis | Pyrrhula erythaca | Pyrrhula erythrocephala | Pyrrhula murina | Pyrrhula leucogenis | Pyrrhula aurantiaca | Pyrrhula pyrrhula | Fringilla montifringilla | Fringilla teydea | Fringilla coelebs | Fringilla polatzeki | Calcarius lapponicus | Calcarius pictus | Calcarius ornatus | Calcarius mccownii | Emberiza variabilis | Emberiza hortulana | Emberiza pusilla | Emberiza cioides | Emberiza tristrami | Emberiza cabanisi | Emberiza jankowskii | Emberiza cineracea | Emberiza rutila | Emberiza sahari | Emberiza buchanani | Emberiza chrysophrys | Emberiza capensis | Emberiza cia | Emberiza schoeniclus | Emberiza stewarti | Emberiza godlewskii | Emberiza pallasi | Emberiza leucocephalos | Emberiza socotrana | Emberiza melanocephala | Emberiza bruniceps | Emberiza koslowi | Emberiza cirlus | Emberiza flaviventris | Emberiza affinis | Emberiza fucata | Emberiza citrinella | Emberiza sulphurata | Emberiza caesia | Plectrophenax nivalis | Plectrophenax hyperboreus | Melophus lathami | Cyanoloxia glaucocaerulea | Cyanoloxia cyanoides | Cyanoloxia rothschildii | Caryothraustes canadensis | Caryothraustes poliogaster | Cyanocompsa brissonii | Cyanocompsa parellina | Passerina caerulea | Passerina cyanea | Passerina amoena | Passerina ciris | Passerina leclancherii | Passerina rositae | Passerina versicolor | Parkerthraustes humeralis | Spiza americana | Pheucticus aureoventris | Pheucticus ludovicianus | Pheucticus melanocephalus | Pheucticus chrysogaster | Pheucticus tibialis | Cardinalis cardinalis | Cardinalis sinuatus | Cardinalis phoeniceus | Porphyrospiza caerulescens | Rhodothraupis celaeno | Periporphyrus erythromelas</span></span>
<span id="cb1-14"><a href="#cb1-14" aria-hidden="true" tabindex="-1"></a><span class="co">#&gt; DateLife query done!</span></span></code></pre></div>
<p>Inspect the structure of the output:</p>
<div class="sourceCode" id="cb2"><pre class="sourceCode r"><code class="sourceCode r"><span id="cb2-1"><a href="#cb2-1" aria-hidden="true" tabindex="-1"></a><span class="fu">str</span>(dquery)</span>
<span id="cb2-2"><a href="#cb2-2" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb2-3"><a href="#cb2-3" aria-hidden="true" tabindex="-1"></a><span class="co">#&gt; List of 3</span></span>
<span id="cb2-4"><a href="#cb2-4" aria-hidden="true" tabindex="-1"></a><span class="co">#&gt;  $ cleaned_names: Named chr [1:289] &quot;Crithagra sulphurata&quot; &quot;Crithagra #&gt; citrinelloides&quot; &quot;Crithagra mozambica&quot; &quot;Crithagra capistrata&quot; ...</span></span>
<span id="cb2-5"><a href="#cb2-5" aria-hidden="true" tabindex="-1"></a><span class="co">#&gt;   ..- attr(*, &quot;names&quot;)= chr [1:289] &quot;Fringillidae1&quot; &quot;Fringillidae2&quot; #&gt; &quot;Fringillidae3&quot; &quot;Fringillidae4&quot; ...</span></span>
<span id="cb2-6"><a href="#cb2-6" aria-hidden="true" tabindex="-1"></a><span class="co">#&gt;  $ ott_ids      : Named int [1:289] 903281 328908 1083721 901359 318589 #&gt; 455546 903271 405220 242220 455541 ...</span></span>
<span id="cb2-7"><a href="#cb2-7" aria-hidden="true" tabindex="-1"></a><span class="co">#&gt;   ..- attr(*, &quot;names&quot;)= chr [1:289] &quot;Crithagra sulphurata&quot; &quot;Crithagra #&gt; citrinelloides&quot; &quot;Crithagra mozambica&quot; &quot;Crithagra capistrata&quot; ...</span></span>
<span id="cb2-8"><a href="#cb2-8" aria-hidden="true" tabindex="-1"></a><span class="co">#&gt;  $ phy          : logi NA</span></span>
<span id="cb2-9"><a href="#cb2-9" aria-hidden="true" tabindex="-1"></a><span class="co">#&gt;  - attr(*, &quot;class&quot;)= chr &quot;datelifeQuery&quot;</span></span></code></pre></div>
<p>We have 289 species names belonging to the family Fringillidae
according to OpenTree’s unified taxonomy. We did not provide an input
tree, so <code>$phy</code> is <code>NA</code>.</p>
</div>
<div id="search-the-species-names-across-a-chronogram-database" class="section level2">
<h2>2. Search the species names across a chronogram database</h2>
<p>Load the prebuilt OpenTree chronogram database object names
<code>opentree_chronograms</code>:</p>
<div class="sourceCode" id="cb3"><pre class="sourceCode r"><code class="sourceCode r"><span id="cb3-1"><a href="#cb3-1" aria-hidden="true" tabindex="-1"></a>utils<span class="sc">::</span><span class="fu">data</span>(<span class="st">&quot;opentree_chronograms&quot;</span>, <span class="at">package =</span> <span class="st">&quot;datelife&quot;</span>)</span></code></pre></div>
<p>Use your <code>datelifeQuery</code> object named <code>dquery</code>
to search the chronogram database:</p>
<div class="sourceCode" id="cb4"><pre class="sourceCode r"><code class="sourceCode r"><span id="cb4-1"><a href="#cb4-1" aria-hidden="true" tabindex="-1"></a>dres <span class="ot">&lt;-</span> datelife<span class="sc">::</span><span class="fu">get_datelife_result</span>(<span class="at">input =</span> dquery, <span class="at">cache =</span> opentree_chronograms)</span>
<span id="cb4-2"><a href="#cb4-2" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb4-3"><a href="#cb4-3" aria-hidden="true" tabindex="-1"></a><span class="co">#&gt; ... Searching the chronogram database.</span></span>
<span id="cb4-4"><a href="#cb4-4" aria-hidden="true" tabindex="-1"></a><span class="co">#&gt; Search done!</span></span>
<span id="cb4-5"><a href="#cb4-5" aria-hidden="true" tabindex="-1"></a><span class="co">#&gt;</span></span>
<span id="cb4-6"><a href="#cb4-6" aria-hidden="true" tabindex="-1"></a><span class="co">#&gt; Input taxon names were found in 19 chronograms.</span></span></code></pre></div>
<p>Results from the chronogram search are returned as a
<code>datelifeResult</code> object, which is a named list of patristic
matrices. Each element of the list corresponds to a chronogram that has
been translated to a distance matrix or patristic matrix. The main
reason for this is that pruning is much faster on a matrix than on a
<code>phylo</code> object:</p>
<div class="sourceCode" id="cb5"><pre class="sourceCode r"><code class="sourceCode r"><span id="cb5-1"><a href="#cb5-1" aria-hidden="true" tabindex="-1"></a><span class="fu">class</span>(dres)</span>
<span id="cb5-2"><a href="#cb5-2" aria-hidden="true" tabindex="-1"></a><span class="co">#&gt; [1] &quot;datelifeResult&quot;</span></span>
<span id="cb5-3"><a href="#cb5-3" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb5-4"><a href="#cb5-4" aria-hidden="true" tabindex="-1"></a><span class="fu">typeof</span>(dres)</span>
<span id="cb5-5"><a href="#cb5-5" aria-hidden="true" tabindex="-1"></a><span class="co">#&gt; [1] &quot;list&quot;</span></span>
<span id="cb5-6"><a href="#cb5-6" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb5-7"><a href="#cb5-7" aria-hidden="true" tabindex="-1"></a><span class="fu">class</span>(dres[[<span class="dv">1</span>]])</span>
<span id="cb5-8"><a href="#cb5-8" aria-hidden="true" tabindex="-1"></a><span class="co">#&gt; [1] &quot;matrix&quot; &quot;array&quot;</span></span></code></pre></div>
<p>Names on your <code>datelifeResult</code> object correspond to study
citations from source chronograms:</p>
<div class="sourceCode" id="cb6"><pre class="sourceCode r"><code class="sourceCode r"><span id="cb6-1"><a href="#cb6-1" aria-hidden="true" tabindex="-1"></a><span class="fu">names</span>(dres)</span>
<span id="cb6-2"><a href="#cb6-2" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb6-3"><a href="#cb6-3" aria-hidden="true" tabindex="-1"></a><span class="co">#&gt; [1] &quot;Hooper, Daniel M., Trevor D. Price. 2017. Chromosomal inversion differences correlate with range overlap in passerine birds. Nature Ecology &amp; Evolution 1 (10): 1526-1534&quot;                                                                                                                                                              </span></span>
<span id="cb6-4"><a href="#cb6-4" aria-hidden="true" tabindex="-1"></a><span class="co">#&gt; [2] &quot;Roquet, C., Lavergne, S., &amp; Thuiller, W. (2014). One Tree to Link Them All: A Phylogenetic Dataset for the European Tetrapoda. PLoS Currents. doi:10.1371/currents.tol.5102670fff8aa5c918e78f5592790e48\n&quot;                                                                                                                              </span></span>
<span id="cb6-5"><a href="#cb6-5" aria-hidden="true" tabindex="-1"></a><span class="co">#&gt; [3] &quot;Roquet, C., Lavergne, S., &amp; Thuiller, W. (2014). One Tree to Link Them All: A Phylogenetic Dataset for the European Tetrapoda. PLoS Currents. doi:10.1371/currents.tol.5102670fff8aa5c918e78f5592790e48\n&quot;                                                                                                                              </span></span>
<span id="cb6-6"><a href="#cb6-6" aria-hidden="true" tabindex="-1"></a><span class="co">#&gt; [4] &quot;Oliveros, C. H., Field, D. J., Ksepka, D. T., Barker, F. K., Aleixo, A., Andersen, M. J., … Faircloth, B. C. (2019). Earth history and the passerine superradiation. Proceedings of the National Academy of Sciences, 116(16), 7916–7925. doi:10.1073/pnas.1813206116\n&quot;                                                                </span></span>
<span id="cb6-7"><a href="#cb6-7" aria-hidden="true" tabindex="-1"></a><span class="co">#&gt; [5] &quot;Kimball, R. T., Oliveros, C. H., Wang, N., White, N. D., Barker, F. K., Field, D. J., … Braun, E. L. (2019). A Phylogenomic Supertree of Birds. Diversity, 11(7), 109. doi:10.3390/d11070109\n&quot;                                                                                                                                         </span></span>
<span id="cb6-8"><a href="#cb6-8" aria-hidden="true" tabindex="-1"></a><span class="co">#&gt; [6] &quot;Kimball, R. T., Oliveros, C. H., Wang, N., White, N. D., Barker, F. K., Field, D. J., … Braun, E. L. (2019). A Phylogenomic Supertree of Birds. Diversity, 11(7), 109. doi:10.3390/d11070109\n&quot;                                                                                                                                         </span></span>
<span id="cb6-9"><a href="#cb6-9" aria-hidden="true" tabindex="-1"></a><span class="co">#&gt; [7] &quot;Uyeda, J. C., Pennell, M. W., Miller, E. T., Maia, R., &amp; McClain, C. R. (2017). The Evolution of Energetic Scaling across the Vertebrate Tree of Life. The American Naturalist, 190(2), 185–199. doi:10.1086/692326\n&quot;                                                                                                                  </span></span>
<span id="cb6-10"><a href="#cb6-10" aria-hidden="true" tabindex="-1"></a><span class="co">#&gt; [8] &quot;Hedges, S. Blair, Julie Marin, Michael Suleski, Madeline Paymer, Sudhir Kumar. 2015. Tree of life reveals clock-like speciation and diversification. Molecular Biology and Evolution 32 (4): 835-845&quot;                                                                                                                                   </span></span>
<span id="cb6-11"><a href="#cb6-11" aria-hidden="true" tabindex="-1"></a><span class="co">#&gt; [9] &quot;Hedges, S. Blair, Julie Marin, Michael Suleski, Madeline Paymer, Sudhir Kumar. 2015. Tree of life reveals clock-like speciation and diversification. Molecular Biology and Evolution 32 (4): 835-845&quot;                                                                                                                                   </span></span>
<span id="cb6-12"><a href="#cb6-12" aria-hidden="true" tabindex="-1"></a><span class="co">#&gt; [10] &quot;Claramunt, Santiago, Joel Cracraft. 2015. A new time tree reveals Earth historys imprint on the evolution of modern birds. Science Advances 1 (11): e1501005-e1501005&quot;                                                                                                                                                                  </span></span>
<span id="cb6-13"><a href="#cb6-13" aria-hidden="true" tabindex="-1"></a><span class="co">#&gt; [11] &quot;Gibb, Gillian C., Ryan England, Gerrit Hartig, P.A. (Trish) McLenachan, Briar L. Taylor Smith, Bennet J. McComish, Alan Cooper, David Penny. 2015. New Zealand passerines help clarify the diversification of major songbird lineages during the Oligocene. Genome Biology and Evolution 7 (11): 2983-2995.&quot;                            </span></span>
<span id="cb6-14"><a href="#cb6-14" aria-hidden="true" tabindex="-1"></a><span class="co">#&gt; [12] &quot;Barker, F. Keith, Kevin J. Burns, John Klicka, Scott M. Lanyon, Irby J. Lovette. 2015. New insights into New World biogeography: An integrated view from the phylogeny of blackbirds, cardinals, sparrows, tanagers, warblers, and allies. The Auk 132 (2): 333-348.&quot;   </span></span>
<span id="cb6-15"><a href="#cb6-15" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb6-16"><a href="#cb6-16" aria-hidden="true" tabindex="-1"></a><span class="co">#&gt; [13] &quot;Barker, F. Keith, Kevin J. Burns, John Klicka, Scott M. Lanyon, Irby J. Lovette. 2015. New insights into New World biogeography: An integrated view from the phylogeny of blackbirds, cardinals, sparrows, tanagers, warblers, and allies. The Auk 132 (2): 333-348.&quot;                                                                   </span></span>
<span id="cb6-17"><a href="#cb6-17" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb6-18"><a href="#cb6-18" aria-hidden="true" tabindex="-1"></a><span class="co">#&gt; [14] &quot;Jetz, W., G. H. Thomas, J. B. Joy, K. Hartmann, A. O. Mooers. 2012. The global diversity of birds in space and time. Nature 491 (7424): 444-448&quot;                                                                                              </span></span>
<span id="cb6-19"><a href="#cb6-19" aria-hidden="true" tabindex="-1"></a><span class="co">#&gt; [15] &quot;Jetz, W., G. H. Thomas, J. B. Joy, K. Hartmann, A. O. Mooers. 2012. The global diversity of birds in space and time. Nature 491 (7424): 444-448&quot;                                                                                                                                                                                        </span></span>
<span id="cb6-20"><a href="#cb6-20" aria-hidden="true" tabindex="-1"></a><span class="co">#&gt; [16] &quot;Barker, F. K., K. J. Burns, J. Klicka, S. M. Lanyon, I. J. Lovette. 2013. Going to extremes: contrasting rates of diversification in a recent radiation of New World passerine birds. Systematic Biology 62 (2): 298-320.&quot;                                                                                                              </span></span>
<span id="cb6-21"><a href="#cb6-21" aria-hidden="true" tabindex="-1"></a><span class="co">#&gt; [17] &quot;Burns, Kevin J., Allison J. Shultz, Pascal O. Title, Nicholas A. Mason, F. Keith Barker, John Klicka, Scott M. Lanyon, Irby J. Lovette. 2014. Phylogenetics and diversification of tanagers (Passeriformes: Thraupidae), the largest radiation of Neotropical songbirds. Molecular Phylogenetics and Evolution 75: 41-77.&quot;              </span></span>
<span id="cb6-22"><a href="#cb6-22" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb6-23"><a href="#cb6-23" aria-hidden="true" tabindex="-1"></a><span class="co">#&gt; [18] &quot;Price, Trevor D., Daniel M. Hooper, Caitlyn D. Buchanan, Ulf S. Johansson, D. Thomas Tietze, Per Alström, Urban Olsson, Mousumi Ghosh-Harihar, Farah Ishtiaq, Sandeep K. Gupta, Jochen Martens, Bettina Harr, Pratap Singh, Dhananjai Mohan. 2014. Niche filling slows the diversification of Himalayan songbirds. Nature 509: 222-225.&quot;</span></span>
<span id="cb6-24"><a href="#cb6-24" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb6-25"><a href="#cb6-25" aria-hidden="true" tabindex="-1"></a><span class="co">#&gt; [19] &quot;Price, Trevor D., Daniel M. Hooper, Caitlyn D. Buchanan, Ulf S. Johansson, D. Thomas Tietze, Per Alström, Urban Olsson, Mousumi Ghosh-Harihar, Farah Ishtiaq, Sandeep K. Gupta, Jochen Martens, Bettina Harr, Pratap Singh, Dhananjai Mohan. 2014. Niche filling slows the diversification of Himalayan songbirds. Nature 509: 222-225.&quot;</span></span></code></pre></div>
</div>
<div id="summarizing-a-dateliferesult-object" class="section level2">
<h2>3. Summarizing a <code>datelifeResult</code> object</h2>
<p>What if you want to plot all your <code>datelifeResult</code>
chronograms individually? What if you want a single tree summarizing all
your source chronograms? The function
<code>datelife::summary.datelifeResult()</code> will get all types of
summaries for you, so you can access them later:</p>
<div class="sourceCode" id="cb7"><pre class="sourceCode r"><code class="sourceCode r"><span id="cb7-1"><a href="#cb7-1" aria-hidden="true" tabindex="-1"></a>dsumm <span class="ot">&lt;-</span> <span class="fu">summary</span>(dres, <span class="at">datelife_query =</span> dquery)</span>
<span id="cb7-2"><a href="#cb7-2" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb7-3"><a href="#cb7-3" aria-hidden="true" tabindex="-1"></a><span class="co">#&gt; Trying with overlap = 2</span></span>
<span id="cb7-4"><a href="#cb7-4" aria-hidden="true" tabindex="-1"></a><span class="co">#&gt;</span></span>
<span id="cb7-5"><a href="#cb7-5" aria-hidden="true" tabindex="-1"></a><span class="co">#&gt; Success!</span></span>
<span id="cb7-6"><a href="#cb7-6" aria-hidden="true" tabindex="-1"></a><span class="co">#&gt;</span></span>
<span id="cb7-7"><a href="#cb7-7" aria-hidden="true" tabindex="-1"></a><span class="co">#&gt; ... Calculating a median summary chronogram.</span></span>
<span id="cb7-8"><a href="#cb7-8" aria-hidden="true" tabindex="-1"></a><span class="co">#&gt;   |==========================================================| 100%</span></span>
<span id="cb7-9"><a href="#cb7-9" aria-hidden="true" tabindex="-1"></a><span class="co">#&gt;   |==========================================================| 100%</span></span>
<span id="cb7-10"><a href="#cb7-10" aria-hidden="true" tabindex="-1"></a><span class="co">#&gt;   |==========================================================| 100%</span></span>
<span id="cb7-11"><a href="#cb7-11" aria-hidden="true" tabindex="-1"></a><span class="co">#&gt; 1 out of 13 chronograms tried:  Ok.</span></span>
<span id="cb7-12"><a href="#cb7-12" aria-hidden="true" tabindex="-1"></a><span class="co">#&gt; 2 out of 13 chronograms tried:  Ok.</span></span>
<span id="cb7-13"><a href="#cb7-13" aria-hidden="true" tabindex="-1"></a><span class="co">#&gt; 3 out of 13 chronograms tried:  Ok.</span></span>
<span id="cb7-14"><a href="#cb7-14" aria-hidden="true" tabindex="-1"></a><span class="co">#&gt; 4 out of 13 chronograms tried:  Ok.</span></span>
<span id="cb7-15"><a href="#cb7-15" aria-hidden="true" tabindex="-1"></a><span class="co">#&gt; 5 out of 13 chronograms tried:  Ok.</span></span>
<span id="cb7-16"><a href="#cb7-16" aria-hidden="true" tabindex="-1"></a><span class="co">#&gt; 6 out of 13 chronograms tried:  Ok.</span></span>
<span id="cb7-17"><a href="#cb7-17" aria-hidden="true" tabindex="-1"></a><span class="co">#&gt; 7 out of 13 chronograms tried:  Ok.</span></span>
<span id="cb7-18"><a href="#cb7-18" aria-hidden="true" tabindex="-1"></a><span class="co">#&gt; 8 out of 13 chronograms tried:  Ok.</span></span>
<span id="cb7-19"><a href="#cb7-19" aria-hidden="true" tabindex="-1"></a><span class="co">#&gt; 9 out of 13 chronograms tried:  Ok.</span></span>
<span id="cb7-20"><a href="#cb7-20" aria-hidden="true" tabindex="-1"></a><span class="co">#&gt; 10 out of 13 chronograms tried:  Ok.</span></span>
<span id="cb7-21"><a href="#cb7-21" aria-hidden="true" tabindex="-1"></a><span class="co">#&gt; 12 out of 19 chronograms tried:  Ok.</span></span>
<span id="cb7-22"><a href="#cb7-22" aria-hidden="true" tabindex="-1"></a><span class="co">#&gt; 13 out of 19 chronograms tried:  Ok.</span></span>
<span id="cb7-23"><a href="#cb7-23" aria-hidden="true" tabindex="-1"></a><span class="co">#&gt; 14 out of 19 chronograms tried:  Ok.</span></span>
<span id="cb7-24"><a href="#cb7-24" aria-hidden="true" tabindex="-1"></a><span class="co">#&gt; 15 out of 19 chronograms tried:  Ok.</span></span>
<span id="cb7-25"><a href="#cb7-25" aria-hidden="true" tabindex="-1"></a><span class="co">#&gt; 16 out of 19 chronograms tried:  Ok.</span></span>
<span id="cb7-26"><a href="#cb7-26" aria-hidden="true" tabindex="-1"></a><span class="co">#&gt; 17 out of 19 chronograms tried:  Ok.</span></span>
<span id="cb7-27"><a href="#cb7-27" aria-hidden="true" tabindex="-1"></a><span class="co">#&gt; 18 out of 19 chronograms tried:  Failed.</span></span>
<span id="cb7-28"><a href="#cb7-28" aria-hidden="true" tabindex="-1"></a><span class="co">#&gt; 19 out of 19 chronograms tried:  Ok.</span></span>
<span id="cb7-29"><a href="#cb7-29" aria-hidden="true" tabindex="-1"></a><span class="co">#&gt;</span></span>
<span id="cb7-30"><a href="#cb7-30" aria-hidden="true" tabindex="-1"></a><span class="co">#&gt;  Synthesizing 18 chronograms with SDM</span></span>
<span id="cb7-31"><a href="#cb7-31" aria-hidden="true" tabindex="-1"></a><span class="co">#&gt;   |==========================================================| 100%</span></span>
<span id="cb7-32"><a href="#cb7-32" aria-hidden="true" tabindex="-1"></a><span class="co">#&gt;   |==========================================================| 100%</span></span>
<span id="cb7-33"><a href="#cb7-33" aria-hidden="true" tabindex="-1"></a><span class="co">#&gt;   |==========================================================| 100%</span></span></code></pre></div>
<p>Now we can explore different age summaries.</p>
<div id="source-chronograms" class="section level3">
<h3>3.1 Source chronograms:</h3>
<p>Chronograms that matched our database search, and that were pruned to
contain only the species names we looked for are called source
chronograms. These can be accessed as follow:</p>
<div class="sourceCode" id="cb8"><pre class="sourceCode r"><code class="sourceCode r"><span id="cb8-1"><a href="#cb8-1" aria-hidden="true" tabindex="-1"></a>dsumm<span class="sc">$</span>phylo_all</span>
<span id="cb8-2"><a href="#cb8-2" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb8-3"><a href="#cb8-3" aria-hidden="true" tabindex="-1"></a><span class="co">#&gt; 19 phylogenetic trees</span></span></code></pre></div>
<p>By study name:</p>
<div class="sourceCode" id="cb9"><pre class="sourceCode r"><code class="sourceCode r"><span id="cb9-1"><a href="#cb9-1" aria-hidden="true" tabindex="-1"></a>dsumm<span class="sc">$</span>phylo_all<span class="sc">$</span><span class="st">`</span><span class="at">Gibb, Gillian C., Ryan England, Gerrit Hartig, P.A. (Trish) McLenachan, Briar L. Taylor Smith, Bennet J. McComish, Alan Cooper, David Penny. 2015. New Zealand passerines help clarify the diversification of major songbird lineages during the Oligocene. Genome Biology and Evolution 7 (11): 2983-2995.</span><span class="st">`</span></span>
<span id="cb9-2"><a href="#cb9-2" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb9-3"><a href="#cb9-3" aria-hidden="true" tabindex="-1"></a><span class="co">#&gt; Phylogenetic tree with 7 tips and 6 internal nodes.</span></span>
<span id="cb9-4"><a href="#cb9-4" aria-hidden="true" tabindex="-1"></a><span class="co">#&gt; </span></span>
<span id="cb9-5"><a href="#cb9-5" aria-hidden="true" tabindex="-1"></a><span class="co">#&gt; Tip labels:</span></span>
<span id="cb9-6"><a href="#cb9-6" aria-hidden="true" tabindex="-1"></a><span class="co">#&gt;   Schoeniclus_spodocephala, Emberiza_tristrami, Emberiza_chrysophrys, #&gt; Emberiza_pusilla, Schoeniclus_aureolus, Serinus_canaria, ...</span></span>
<span id="cb9-7"><a href="#cb9-7" aria-hidden="true" tabindex="-1"></a><span class="co">#&gt; </span></span>
<span id="cb9-8"><a href="#cb9-8" aria-hidden="true" tabindex="-1"></a><span class="co">#&gt; Rooted; includes branch lengths.</span></span></code></pre></div>
<p>Or, by index position:</p>
<div class="sourceCode" id="cb10"><pre class="sourceCode r"><code class="sourceCode r"><span id="cb10-1"><a href="#cb10-1" aria-hidden="true" tabindex="-1"></a><span class="fu">names</span>(dsumm<span class="sc">$</span>phylo_all)[<span class="dv">1</span>]</span>
<span id="cb10-2"><a href="#cb10-2" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb10-3"><a href="#cb10-3" aria-hidden="true" tabindex="-1"></a><span class="co">#&gt; [1] &quot;Hooper, Daniel M., Trevor D. Price. 2017. Chromosomal inversion differences correlate with range overlap in passerine birds. Nature Ecology &amp; Evolution 1 (10): 1526-1534&quot;</span></span>
<span id="cb10-4"><a href="#cb10-4" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb10-5"><a href="#cb10-5" aria-hidden="true" tabindex="-1"></a>dsumm<span class="sc">$</span>phylo_all[[<span class="dv">1</span>]]</span>
<span id="cb10-6"><a href="#cb10-6" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb10-7"><a href="#cb10-7" aria-hidden="true" tabindex="-1"></a><span class="co">#&gt; Phylogenetic tree with 44 tips and 43 internal nodes.</span></span>
<span id="cb10-8"><a href="#cb10-8" aria-hidden="true" tabindex="-1"></a><span class="co">#&gt; </span></span>
<span id="cb10-9"><a href="#cb10-9" aria-hidden="true" tabindex="-1"></a><span class="co">#&gt; Tip labels:</span></span>
<span id="cb10-10"><a href="#cb10-10" aria-hidden="true" tabindex="-1"></a><span class="co">#&gt;   Peucedramus_taeniatus, Peucaea_botterii, Schoeniclus_spodocephala, #&gt; Schoeniclus_aureolus, Emberiza_chrysophrys, Emberiza_pusilla, ...</span></span>
<span id="cb10-11"><a href="#cb10-11" aria-hidden="true" tabindex="-1"></a><span class="co">#&gt; </span></span>
<span id="cb10-12"><a href="#cb10-12" aria-hidden="true" tabindex="-1"></a><span class="co">#&gt; Rooted; includes branch lengths.</span></span></code></pre></div>
</div>
</div>



<!-- code folding -->


<!-- dynamically load mathjax for compatibility with self-contained -->
<script>
  (function () {
    var script = document.createElement("script");
    script.type = "text/javascript";
    script.src  = "https://mathjax.rstudio.com/latest/MathJax.js?config=TeX-AMS-MML_HTMLorMML";
    document.getElementsByTagName("head")[0].appendChild(script);
  })();
</script>

</body>
</html>
